pre-check/v1.1.x/check.sh

```diff
@@ -305,6 +305,27 @@ check_free_space()
     record_fail
 }
 
+check_certificate()
+{
+    echo ">>> Check certificates..."
+
+    cert=$(kubectl get settings.harvesterhci.io ssl-certificates -o jsonpath='{.value}' | jq -r '.publicCertificate')
+    if [ -z "$cert" ]; then
+        echo "ssl-certificates is not set."
+    else
+        echo "Checking if the certificate is a self-signed one..."
+        subject=$(echo "$cert" | openssl x509 -noout -inform PEM -subject | sed 's/subject=//')
+        issuer=$(echo "$cert" | openssl x509 -noout -inform PEM -issuer | sed 's/issuer=//')
+        if [ "$subject" = "$issuer" ]; then
+            echo "Self-signed certificate found!"
+            echo "For more information, please visit https://github.com/harvester/harvester/issues/4519"
```

Comment:
I think the preferred check method would be:
1.  Use openssl to extract subject alternative names from the certificate, e.g.,
    ```
    SAN=$(openssl x509 -in "$CERTIFICATE_FILE" -noout -text | grep -oP '(?<=DNS:|IP Address:)[^,]+')
    ```

1.  Check if the target VIP IP is in the list of subject alternative names
    ```
    if echo "$SAN" | grep -q "$VIP"; then
        echo "The certificate contains the VIP address $VIP"
    else
        echo "The certificate does not contain the VIP address $VIP"
        record_fail & echo a proper error message
    fi
    ```